Cython/Compiler/Nodes.py

```diff
@@ -815,6 +815,7 @@ class CArgDeclNode(Node):
     # kw_only        boolean            Is a keyword-only argument
     # is_dynamic     boolean            Non-literal arg stored inside CyFunction
     # pos_only       boolean            Is a positional-only argument
+    # type_set_manually boolean         Used in overriding the type of self arguments - otherwise unimportant
```

Comment:
I'm not convinced of this feature yet. I think it's worth a separate discussion. There might be other ways to achieve the original intention, for example.

Reply:
I'm not convinced it should be possible to override the type of self. Being able to call a type's method on something that is not the type? How is that not a static method then?

Reply:
The main counter-argument is that it is possible in Python 3 where functions and methods are the same thing.

The difference between it and a static method is that in staticmethod always drops the `self` argument whether you call it with an instance or not, but it is acting very similar to a static method.

I definitely don't think this would be a good thing to do regularly. @jdemeyer [claimed to have use cases for it](https://github.com/cython/cython/issues/1274#issuecomment-525230079) so maybe could suggest a good reason? I'd mainly implemented it here because it seemed like a necessary step for the arbitrary decorator support.